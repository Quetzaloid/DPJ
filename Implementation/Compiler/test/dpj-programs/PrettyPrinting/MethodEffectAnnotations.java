/**
 * Method effect annotations
 * 
 * @author Rob Bocchino
 */
public class MethodEffectAnnotations {
    region r1, r2;
    public MethodEffectAnnotations() {}
    public MethodEffectAnnotations(int x) pure {}
    public MethodEffectAnnotations(char x) reads r1, r2 writes r1, r2 {}
    void m1() {}
    void m2() pure {}
    void m3() reads r1, r2 writes r1, r2 {}
    commutative void m4() {}
}
