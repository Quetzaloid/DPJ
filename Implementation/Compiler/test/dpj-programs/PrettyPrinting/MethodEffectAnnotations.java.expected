
public class MethodEffectAnnotations {
    region r1;
    region r2;
    
    public MethodEffectAnnotations() {
        super();
    }
    
    public MethodEffectAnnotations(int x) pure {
        super();
    }
    
    public MethodEffectAnnotations(char x) reads r1, r2 writes r1, r2 {
        super();
    }
    
    void m1() {
    }
    
    void m2() pure {
    }
    
    void m3() reads r1, r2 writes r1, r2 {
    }
    
    commutative void m4() {
    }
}